### PR TITLE
Add Janet support

### PR DESCRIPTION
**Pull request by sogaiu**
This PR is an attempt to add support for the [Janet programming language](https://janet-lang.org/), a language with a lisp-ish / clojure-ish syntax.

In Janet, mutable collections are distinguished from immutable ones via a leading `@`.  In the tree-sitter grammar used in this PR, `@{`, `@(`, and `@[` are tokens indicating opening delimiters.  I hope the following code in `src/tree_sitter_parser.rs` was appropriate:
```
delimiter_tokens: (vec![("@{", "}"), ("@(", ")"), ("@[", "]"),
                        ("{", "}"), ("(", ")"), ("[", "]")])
```

### CODE REVIEW

**Review by Wilfred — CHANGES_REQUESTED** [review]
Other than the display name, this looks pretty good :)

I just need two changes before merging:

(1) Run the formatter (you can run `cargo fmt`).

(2) Add a sample file to `sample_files/` and run `./sample_files/compare_all.sh; cp sample_files/compare.result sample_files/compare.expected`. That way we can see everything is set up correctly and will stay working in future.

File: src/tree_sitter_parser.rs [hunk 1]
```diff
@@ -273,6 +274,23 @@ pub fn from_language(language: guess::Language) -> TreeSitterConfig {
                 .unwrap(),
             }
         }
+        JanetSimple => {
+            let language = unsafe { tree_sitter_janet_simple() };
+            TreeSitterConfig {
+                name: "Janet Simple",
```
[the remark this answers is not in the document]
Comment (sogaiu) [reply src/tree_sitter_parser.rs:280]: Just to give some background, there is [another tree-sitter grammar for Janet](https://github.com/GrayJack/tree-sitter-janet) so my intent was to leave room for it.

Should I still change the name to "Janet"?

[the remark this answers is not in the document]
Comment (Wilfred) [reply src/tree_sitter_parser.rs:280]: Yes please -- I think it's unlikely difftastic will end up with multiple parsers for the same language, and this feels like an implementation detail.

[the remark this answers is not in the document]
Comment (sogaiu) [reply src/tree_sitter_parser.rs:280]: Thanks for the clarification.

**Comment by sogaiu** [comment]
@Wilfred I've made attempts to address:

* changing the display name to "Janet"
* formatting the code with `cargo fmt`
* adding sample files and updating `compare.expected`

Please let me know if those went ok and/or there is anything else.

**Comment by Wilfred** [comment]
Thanks!

**Pull request merged**